docs/guides/dev_guide.rst

```diff
@@ -407,8 +407,186 @@ conftest.py. If they're generic/useful enough consider putting them into
 one of the `fixtures/` directory for use in `cfme_tests` or the `plugin/`
 directory for use in both projects.
 
+Requirements workflow
+^^^^^^^^^^^^^^^^^^^^^
+We have recently updated our requirements workflow in the hope that it provides a more
+streamlined experience. The requirements workflow revolves around the ``miq requirement``
+command:
+
+.. code-block:: console
+
+    Usage: miq requirement [OPTIONS] COMMAND [ARGS]...
+
+      Functions for adding, updating, and freezing requirements
+
+    Options:
+      --help  Show this message and exit.
+
+    Commands:
+      add          Add and install/update a package to current virtualenv.
```

Comment:
What happens if a package fails to install? does it still get added?

Reply:
@kedark3 it does not, here's the chunk of code that actually install the package. 
``` python,

   # try to install the package
    if upgrade_or_not:
        commands = ["pip", "install", "--upgrade", package_name]
    else:
        commands = ["pip", "install", package_name]
    process = subprocess.run(commands)
    if process.returncode != 0:
        click.echo(f"Unable to install {package_name}")
        sys.exit(1)
```

So if the process fails then we exit and the package name isn't added to any file.

Reply:
That looks perfect. thanks.